src/Moderation/ReasonDialog.jsx

```diff
@@ -0,0 +1,150 @@
+const type = props.type;
+const open = props.open;
+const closeModal = props.closeModal;
+const submitClick = props.submitClick;
+
+const submitAndClose = (reason) => {
+  submitClick(reason);
+  closeModal();
+};
+
+const buildContentReportOptions = () => {
+  const reasons = {
+    spam: "Spam",
+    abuse: "Hate, Bullying or Harassment",
+    explicit: "Nudity or Suggestive Content",
+    fraud: "Frauds or Scams",
+    illegal_content: "Illegal Content",
+    copyright_violation: "Copyright Violation",
+    other: "Other",
+  };
+
+  const descriptions = {
+    spam: "Repeated, unwanted, or unsolicited actions.",
+    abuse: "Discrimination, threats, or targeted intimidation.",
+    explicit: "Explicit or inappropriate visuals or text.",
+    fraud: "Deceptive practices or misleading offers.",
+    illegal_content: "Promotion or distribution of illegal activities.",
+    copyright_violation: "Unauthorized use of intellectual property.",
+    other: "Content that doesn't fit the above categories.",
+  };
+
+  return buildDialog(reasons, descriptions);
+};
+
+const buildAccountReportOptions = () => {
+  const reasons = {
+    bot: "Bot or Automated Actions",
+    impersonation: "Impersonation",
+    inappropriate_profile: "Inappropriate Profile Information",
+    posting_inappropriate_content: "Posting Inappropriate Content",
+    other: "Other",
+  };
+
+  const descriptions = {
+    bot: "Non-human activities or automated interactions.",
+    impersonation: "Posing as another individual or a notable entity.",
+    inappropriate_profile: "Deceptive bios, inappropriate profile visuals, or misleading details.",
+    posting_inappropriate_content: "Continuous sharing of explicit, misleading, or harmful content.",
+    other: "Issues not addressed by the above categories.",
+  };
+
+  return buildDialog(reasons, descriptions);
+};
+
+const buildDialog = (reasons, descriptions) => {
+  return (
+    <div className="d-flex flex-column gap-3">
+      <div className="d-flex flex-column gap-3">
+        <div className="d-flex flex-row gap-2">
+          <h4>Report {type}</h4>
+        </div>
+        <div className="d-flex flex-row gap-2">
+          <span>Tell us more about why this {type} is problematic.</span>
+        </div>
+      </div>
+      <div className="d-flex flex-column gap-3">
+        {Object.keys(reasons).map((reason) => {
+          return (
+            <div className="d-flex flex-row gap-2">
```

Comment:
Please provide an unique `key` for the element:

```suggestion
            <div className="d-flex flex-row gap-2" key={reason.id // or what is the unique value here}>
```

Reply:
React isn't complaining about this for me since it's not a React component. Why do we need a key?